qiskit_experiments/library/characterization/cr_hamiltonian.py

```diff
@@ -0,0 +1,472 @@
+
+from qiskit_experiments.framework import BaseExperiment, Options
+from .cr_hamiltonian_analysis import CrossResonanceHamiltonianAnalysis
+
+
+class CrossResonanceHamiltonian(BaseExperiment):
+    r"""Cross resonance Hamiltonian tomography experiment.
+
+    # section: overview
+
+        This experiment assumes the two qubit Hamiltonian in the form
+
+        .. math::
+
+            H = \frac{I \otimes A}{2} + \frac{Z \otimes B}{2}
+
+        where :math:`A` and :math:`B` are Pauli operator :math:`\in {X, Y, Z}`.
+        The coefficient of each Pauli term in the Hamiltonian
+        can be estimated with this experiment.
+
+        This experiment is performed by stretching a pulse duration of the cross resonance tone
+        and measuring the target qubit by projecting onto x, y, and z basis.
+        We will observe the control qubit state dependent (controlled-) Rabi oscillation on the
+        target qubit. The fit for this oscillation topographically reconstructs the
+        Hamiltonian in the form shown above.
+        Thus the control qubit is prepared in both ground and excited state and
+        the experiment is repeated for both control qubit states.
+        See Ref. [1] for more details.
+
+        More specifically following circuits are executed in this experiment.
+
+        .. parsed-literal::
+
+            (X measurement)
+
+                 ┌───┐┌────────────────────┐
+            q_0: ┤ X ├┤0                   ├────────
+                 └───┘│  cr_tone(duration) │┌───┐┌─┐
+            q_1: ─────┤1                   ├┤ H ├┤M├
+                      └────────────────────┘└───┘└╥┘
+            c: 1/═════════════════════════════════╩═
+                                                  0
+
+            (Y measurement)
+
+                 ┌───┐┌────────────────────┐
+            q_0: ┤ X ├┤0                   ├───────────────
+                 └───┘│  cr_tone(duration) │┌─────┐┌───┐┌─┐
+            q_1: ─────┤1                   ├┤ Sdg ├┤ H ├┤M├
+                      └────────────────────┘└─────┘└───┘└╥┘
+            c: 1/════════════════════════════════════════╩═
+                                                         0
+
+            (Z measurement)
+
+                 ┌───┐┌────────────────────┐
+            q_0: ┤ X ├┤0                   ├───
+                 └───┘│  cr_tone(duration) │┌─┐
+            q_1: ─────┤1                   ├┤M├
+                      └────────────────────┘└╥┘
+            c: 1/════════════════════════════╩═
+                                             0
+
+        The ``X`` gate on the control qubit (``q_0``) depends on the required control state.
+        Here ``cr_tone`` is implemented by a single cross resonance tone
+        driving the control qubit at the frequency of target qubit.
+        The pulse envelope is the flat-topped Gaussian implemented by the parametric pulse
+        :py:class:`~qiskit.pulse.library.parametric_pulses.GaussianSquare`.
```

Comment:
Do we need this restriction here? What if I use a different pulse?

Reply:
Good question. That was really hard decision. 

When we look at the main circuit generation code, there is a logic to scale duration and sigma values with dt and SI prefix. If we allow user to define arbitrary pulse shape, indeed, we need to introduce very complicated logic to infer timing related parameters from pulse parameters. Though it is possible at the price of readability, I decided to follow KISS principle rather than ensure generality (I think most of users are happy with `GaussianSquare`).
https://github.com/Qiskit/qiskit-experiments/blob/1cd936bfc5a8b9558fb79351fe63281bb631aea9/qiskit_experiments/library/characterization/cr_hamiltonian.py#L261-L263

Reply:
OK. I have not run enough CR tomo to have an opinion on how limiting this is. @wshanks any thoughts? I wonder if it makes any sense to have an abstract base class for CR Hamiltonian tomo that does not have this restriction. This would at least make it easy to extend if we want to support other pulse shapes.

Reply:
Abstract base class make sense, but probably this is no longer fit in with characterization module. We create dedicated module for Hamiltonian tomography?

Reply:
Thanks Will. I also think GaussianSquare is enough assumption for now (we can update later).

> allow a custom schedule block to be set with a free duration parameter

That was in the original PR, but I find some edge case
```python
duration = Parameter("duration")

cr_sched = ScheduleBlock()
cr_sched += Play(GaussianSquare(duration, amp=0.1, sigma=64, risefall_sigma_ratio=2), ControlChannel(0))

# case 1
exp = CrossResonanceHamiltonian(qubits=(0, 1), durations=np.linspace(100, 1000), schedule=cr_sched)
exp.set_experiment_options(amp=0.2)  # this doesn't change amp parameter

# case 2
exp = CrossResonanceHamiltonian(qubits=(3, 4), durations=np.linspace(100, 1000), schedule=cr_sched)  # this doesn't apply pulse to qubit 3-4 because control channel is hard coded in given schedule.
```

To avoid these confusion, I dropped an interface for schedule.

Reply:
Sounds good. So we stick to GaussianSquare for this PR.